src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -151,79 +138,28 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
-                    .Where(TableQuery.GenerateFilterCondition("OwningTimeoutManager", QueryComparisons.Equal, endpointName));
-            }
+                            TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual, now.ToString(partitionKeyScope))),
+                        TableOperators.And,
+                        TableQuery.GenerateFilterCondition("OwningTimeoutManager", QueryComparisons.Equal, endpointName))
+                );
 
             var timeoutDataEntities = await timeoutDataTable.ExecuteQueryAsync(query, take: 1000).ConfigureAwait(false);
-            var result = timeoutDataEntities.OrderBy(c => c.Time);
-
-            var allTimeouts = result.ToList();
-            if (allTimeouts.Count == 0)
-            {
-                return new TimeoutsChunk(new TimeoutsChunk.Timeout[0], now.AddSeconds(1));
-            }
 
-            var pastTimeouts = allTimeouts.Where(c => c.Time > startSlice && c.Time <= now).ToList();
-            var futureTimeouts = allTimeouts.Where(c => c.Time > now).ToList();
+            var timeouts = timeoutDataEntities.Where(c => c.Time <= now).OrderBy(c => c.Time).ToList();
 
-            if (lastSuccessfulReadEntity != null)
-            {
-                var catchingUp = lastSuccessfulRead.Value.AddSeconds(catchUpInterval);
-                lastSuccessfulRead = catchingUp > now ? now : catchingUp;
-                lastSuccessfulReadEntity.LastSuccessfullRead = lastSuccessfulRead.Value;
-            }
-
-            var future = futureTimeouts.SafeFirstOrDefault();
-            var nextTimeToRunQuery = lastSuccessfulRead ?? (future?.Time ?? now.AddSeconds(1));
-
-            var timeoutsChunk = new TimeoutsChunk(
-                pastTimeouts.Where(c => !string.IsNullOrEmpty(c.RowKey))
-                    .Select(c => new TimeoutsChunk.Timeout(c.RowKey, c.Time))
-                    .Distinct(new TimoutChunkComparer())
-                    .ToArray(),
-                nextTimeToRunQuery);
+            var dueTimeouts = timeouts.Where(c => !string.IsNullOrEmpty(c.RowKey))
+                .Select(c => new TimeoutsChunk.Timeout(c.RowKey, c.Time))
+                .Distinct(new TimoutChunkComparer())
+                .ToArray();
 
-            updateSuccessfulReadOperationForNextSpin = GetUpdateSuccessfulRead(lastSuccessfulReadEntity);
-            return timeoutsChunk;
-        }
-
-        async Task TryUpdateSuccessfulRead(CloudTable timeoutManagerDataTable)
-        {
-            if (updateSuccessfulReadOperationForNextSpin != null)
-            {
-                try
-                {
-                    await UpdateSuccessfulRead(timeoutManagerDataTable, updateSuccessfulReadOperationForNextSpin).ConfigureAwait(false);
-                }
-                finally
-                {
-                    updateSuccessfulReadOperationForNextSpin = null;
-                }
-            }
+            return new TimeoutsChunk(dueTimeouts, now.AddSeconds(1));
```

Comment:
This is a challenging change. In the past, we did the following
```
var nextTimeToRunQuery = lastSuccessfulRead ?? (future?.Time ?? now.AddSeconds(1));
```

Now we're dropping the future look-up and forcing a next execution of timeout persister to get next chunks every second. Considering that an endpoint is scaled out, this can quickly bring to endpoint been throttled.

Perhaps we should still query a bit into future, for something like next minute or five, and set the `nextTimeToRunQuery` to either the first found future timeout in the queried data, or fallback to the future time we used for query (1 or 5 mins).

```... TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual, now.AddMinutes(futureLookUpTime).ToString(partitionKeyScope)))```

This will substantially reduce number of roundtrips made to the storage.